network.go

```diff
@@ -402,65 +515,68 @@ func (n *network) Delete() error {
 	defer func() {
 		if err != nil {
 			n.dbExists = false
-			if e := ctrlr.updateToStore(n); e != nil {
+			if e := n.getController().updateToStore(n); e != nil {
 				log.Warnf("failed to recreate network in store %s : %v", n.name, e)
 			}
 		}
 	}()
 
-	if err = n.deleteNetwork(); err != nil {
-		return err
-	}
-
 	n.ipamRelease()
 
 	return nil
 }
 
 func (n *network) deleteNetwork() error {
-	n.Lock()
-	id := n.id
-	d := n.driver
-	n.ctrlr.Lock()
-	delete(n.ctrlr.networks, id)
-	n.ctrlr.Unlock()
-	n.Unlock()
+	d, err := n.driver()
+	if err != nil {
+		return fmt.Errorf("failed deleting network: %v", err)
+	}
+
+	// If it is local scope make sure we call the driver about the network
+	// because the network may have been created in some past life of libnetwork.
+	if n.DataScope() == datastore.LocalScope {
```

Comment:
I understand the intention. since this is added specifically to overcome the bridge driver persistence issue, can you change it to actually check for `n.Type() == bridge` ? That way it is clear and we can remove it once we have the persistence logic in bridge driver ?